recipes/caf/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+                self.settings.arch == 'x86':
+            raise ConanInvalidConfiguration("clang >= 11.0 does not support x86")
+        if self.options.shared and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Shared libraries are not supported on Windows")
+        if self.options.with_openssl and self.settings.os == "Windows" and self.settings.arch == "x86":
+            raise ConanInvalidConfiguration("OpenSSL is not supported for Windows x86")
+
+    def _cmake_configure(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["CMAKE_CXX_STANDARD"] = self._cppstd
+            if Version(self.version) <= "0.17.6":
+                self._cmake.definitions["CAF_NO_AUTO_LIBCPP"] = True
+                self._cmake.definitions["CAF_NO_OPENSSL"] = not self.options.with_openssl
+                for define in ["CAF_NO_EXAMPLES", "CAF_NO_TOOLS", "CAF_NO_UNIT_TESTS", "CAF_NO_PYTHON"]:
+                    self._cmake.definitions[define] = "ON"
+                self._cmake.definitions["CAF_BUILD_STATIC"] = not self.options.shared
+                self._cmake.definitions["CAF_BUILD_STATIC_ONLY"] = not self.options.shared
+            else:
+                self._cmake.definitions["CAF_ENABLE_OPENSSL_MODULE"] = self.options.with_openssl
+                for define in ["CAF_ENABLE_EXAMPLES", "CAF_ENABLE_TOOLS", "CAF_ENABLE_TESTING"]:
+                    self._cmake.definitions[define] = "OFF"
+            self._cmake.definitions["CAF_LOG_LEVEL"] = self.options.log_level.value.upper()
+            if self.settings.compiler == 'clang':
+                self._cmake.definitions["PTHREAD_LIBRARIES"] = "-pthread -ldl"
+            else:
+                self._cmake.definitions["PTHREAD_LIBRARIES"] = "-pthread"
+                if self.settings.compiler == "gcc" and Version(self.settings.compiler.version.value) < "5.0":
+                    self._cmake.definitions["CMAKE_SHARED_LINKER_FLAGS"] = "-pthread"
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        cmake = self._cmake_configure()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
+        cmake = self._cmake_configure()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        suffix = "_static" if not self.options.shared and Version(self.version) <= "0.17.6" else ""
+
+        self.cpp_info.names["cmake_find_package"] = "CAF"
+        self.cpp_info.names["cmake_find_package_multi"] = "CAF"
+
+        self.cpp_info.components["core"].names["pkg_config"] = "caf_core{}".format(suffix)
```

Comment:
do you have a link fot these pkgconfig names?

What are CMake imported target names?

Reply:
removed